task/banking.ts

```diff
@@ -0,0 +1,164 @@
+// defining interfaces.
+interface User {
+    id: number;
+    name: string;
+    email: string;
+    password: string;
+}
+
+interface Transaction {
+    id: number;
+    userId: number;
+    amount: number;
+    type: 'withdrawal' | 'deposit' | 'transfer';
+}
+
+interface Account {
+    id: number;
+    userId: number;
+    balance: number;
+}
+
+// defining class for different operations
+class Bank {
+    private users: User[] = [];
+    private transactions: Transaction[] = [];
+    private accounts: Account[] = [];
+
+    createUser(user: User): void {
+        const newUser: User = {
+            id: user.id,
+            name: user.name,
+            email: user.email,
+            password: user.password,
+        };
+        this.users.push(newUser);
+    }
+
+    createAccount(userId: number): void {
+        const newAccount: Account = {
+            id: this.accounts.length + 1,
+            userId,
+            balance: 0,
+        };
+        this.accounts.push(newAccount);
+    }
+
+
+    closeAccount(accountId: number): void {
+        const account = this.accounts.find(key => key.id === accountId);
+        if (account) {
+            if (account.balance >= 0) {
+                console.log(`Account ID ${accountId} has been closed.`);
+            } else {
+                console.log("Account balance is negative. Account cannot be closed.");
+            }
+        } else {
+            console.log("Account not found.");
+        }
+    }
+
+
+    withdraw(accountId: number, amount: number): void {
+        const account = this.accounts.find(key => key.id === accountId);
+        if (account) {
+            if (account.balance >= amount) {
+                account.balance -= amount;
+                const transaction: Transaction = {
+                    id: this.transactions.length + 1,
+                    userId: account.userId,
+                    amount,
+                    type: 'withdrawal',
+                };
+                this.transactions.push(transaction);
+            } else {
+                console.log("Insufficient balance");
+            }
+        } else {
+            console.log("Account not found");
+        }
+    }
+
+
+    deposit(accountId: number, amount: number): void {
+        const account = this.accounts.find(key => key.id === accountId);
+        if (account) {
+            account.balance += amount;
+            const transaction: Transaction = {
+                id: this.transactions.length + 1,
+                userId: account.userId,
+                amount,
+                type: 'deposit',
+            };
+            this.transactions.push(transaction);
```

Comment:
you can directly push the `transaction` parameter into the transactions array instead of creating a new object transaction.